File: 20-29/23/main.js
function getProperDivisorsSum(n) {
    let sum = 1;
    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) {
            sum += i;
            if (i !== n / i) {
                sum += n / i;
            }
        }
    }
    return sum;
}

function findAbundantNumbers(limit) {
    let abundantNumbers = [];
    for (let n = 12; n <= limit; n++) {
        if (getProperDivisorsSum(n) > n) {
            abundantNumbers.push(n);
        }
    }
    return abundantNumbers;
}

const LIMIT = 28123;
const abundantNumbers = findAbundantNumbers(LIMIT);
const abundantSet = new Set(abundantNumbers);

function isSumOfTwoAbundant(n) {
    for (let a of abundantNumbers) {
        if (a >= n) break;
        if (abundantSet.has(n - a)) return true;
    }
    return false;
}

let totalSum = 0;
for (let n = 1; n <= LIMIT; n++) {
    if (!isSumOfTwoAbundant(n)) {
        totalSum += n;
    }
}

console.log("The sum of all positive integers that cannot be written as the sum of two abundant numbers is:", totalSum);

File: 20-29/23/Main.kt
fun getProperDivisorsSum(n: Int): Int {
    var sum = 1
    for (i in 2..Math.sqrt(n.toDouble()).toInt()) {
        if (n % i == 0) {
            sum += i
            if (i != n / i) {
                sum += n / i
            }
        }
    }
    return sum
}

fun findAbundantNumbers(limit: Int): List<Int> {
    val abundantNumbers = mutableListOf<Int>()
    for (n in 12..limit) {
        if (getProperDivisorsSum(n) > n) {
            abundantNumbers.add(n)
        }
    }
    return abundantNumbers
}

fun isSumOfTwoAbundant(n: Int, abundantNumbers: List<Int>, abundantSet: Set<Int>): Boolean {
    for (a in abundantNumbers) {
        if (a >= n) break
        if ((n - a) in abundantSet) {
            return true
        }
    }
    return false
}

fun main() {
    val LIMIT = 28123
    val abundantNumbers = findAbundantNumbers(LIMIT)
    val abundantSet = abundantNumbers.toSet()

    var totalSum = 0
    for (n in 1..LIMIT) {
        if (!isSumOfTwoAbundant(n, abundantNumbers, abundantSet)) {
            totalSum += n
        }
    }

    println("The sum of all positive integers that cannot be written as the sum of two abundant numbers is: $totalSum")
}
